src/fillbroadcast.jl

```diff
@@ -200,11 +202,12 @@ for op in (:+, :-)
     @eval begin
         function broadcasted(::DefaultArrayStyle{1}, ::typeof($op), a::AbstractRange{T}, b::ZerosVector{V}) where {T,V}
             broadcast_shape(axes(a), axes(b)) == axes(a) || throw(ArgumentError("Cannot broadcast $a and $b. Convert $b to a Vector first."))
-            _copy_oftype(a, promote_type(T,V))
+            TT = typeof($op(zero(T), zero(V)))
+            _copy_oftype(a, TT)
         end
         function broadcasted(::DefaultArrayStyle{1}, ::typeof($op), a::AbstractVector{T}, b::ZerosVector{V}) where {T,V}
             broadcast_shape(axes(a), axes(b)) == axes(a) || throw(ArgumentError("Cannot broadcast $a and $b. Convert $b to a Vector first."))
-            TT = promote_type(T,V)
+            TT = typeof($op(zero(T), zero(V)))
             broadcasted(TT, a)
```

Comment:
I'm curious, why is `broadcasted` used here but `_copy_oftype` in the definition of ranges above? Couldn't one use `_copy_oftype` in both cases?

And would it be reasonable to save allocations and only copy if `T !== TT`?

Reply:
It seems `_copy_oftype` would fix the AD issues.